### PR TITLE
Adding flowexp_blackoil

### PR DESCRIPTION
**Pull request by GitPaean**
It is modified from the original ebos. 

It is a lighter weight simulator. It can used to test different features without adapting the main flow simulator, which contains much more development while not as easy to adapt. 

The recently developed compositional simulator is based on this flowexp_blackoil simulator.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

File: flowexperimental/flowexp.hpp [hunk 1]
```diff
@@ -25,8 +25,8 @@
  *
  * \brief The common settings for all ebos variants.
```
Comment (akva2) [inline flowexperimental/flowexp.hpp:26]: should probably clean out references to ebos in this file (comments).

Comment (GitPaean) [reply flowexperimental/flowexp.hpp:26]: Sure. Tried while missed this one.

**Comment by GitPaean** [comment]
jenkins build this update_data please

File: flowexperimental/flowexp.hpp (hunk 1 above)
Comment (akva2) [reply flowexperimental/flowexp.hpp:26]: there are more.

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1156 please

**Comment by GitPaean** [comment]
I am marking this PR as ready for review.

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1156 please

File: flowexperimental/flownewtonmethod.hpp [hunk 2]
```diff
@@ -0,0 +1,266 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
```
Comment (akva2) [inline flowexperimental/flownewtonmethod.hpp:1]: this file should be called flowexpnewtonmethod (and class named accordingly). as is you'd think it was used by flow.

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1156 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1156 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1156 please

**Review by totto82 — APPROVED** [review]
Thanks. I welcome the experimental simulator back.

**Comment by GitPaean** [comment]
I am performing another jenkins test running and planning to get this PR in before the end of the day. Please voice out if you have objections or suggestions.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1156 please

**Comment by GitPaean** [comment]
As announced previously, I am getting this PR in now favoring the later development. And also, this PR does not change any existing code while only adding new code.  The new code will probably will revised or refactoring for improvement along the way, while it does not intervene the main development of OPM-flow. 

It is a play ground for experiments, and it reveal issues with how things work currently in OPM-flow and proposal for refactoring of OPM-flow will likely come.

**Pull request merged**